lib/stream/leo-stream.js

```diff
@@ -1619,7 +1735,49 @@ module.exports = function(configure) {
 
 									var file = item.s3;
 									file.range = `bytes=${fileOffset}-${fileEnd}`;
-									return ls.fromS3(file);
+								}
+							};
+
+							let create_s3_stream = (item) => {
+								if (item.s3 && item.end.localeCompare(start) > 0) {
+
+									var file = item.s3;
+									let bucket = file.bucket || file.Bucket;
+									let key = file.key || file.Key;
+
+									file.uncompressed = opts.fast_s3_read_download_as_uncompressed == true;
+
+									file.localFilename = s3LocalFileHelper.buildLocalFilePath(file, item.end);
+
+									let res;
+									let rej;
+									file.localFileReady = new Promise((resolve, reject) => {
+										res = resolve;
```

Comment:
@jgrantr is there a better way to do this?  I just need a promise that I'll resolve somewhere else